datasets/ambig_qa/ambig_qa.py

```diff
@@ -0,0 +1,151 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""AmbigQA: Answering Ambiguous Open-domain Questions"""
+
+from __future__ import absolute_import, division, print_function
+
+import json
+import os
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{ min2020ambigqa,
+    title={ {A}mbig{QA}: Answering Ambiguous Open-domain Questions },
+    author={ Min, Sewon and Michael, Julian and Hajishirzi, Hannaneh and Zettlemoyer, Luke },
+    booktitle={ EMNLP },
+    year={2020}
+}
+"""
+
+_DESCRIPTION = """\
+AmbigNQ, a dataset covering 14,042 questions from NQ-open, an existing open-domain QA benchmark. We find that over half of the questions in NQ-open are ambiguous. The types of ambiguity are diverse and sometimes subtle, many of which are only apparent after examining evidence provided by a very large text corpus.  AMBIGNQ, a dataset with
+14,042 annotations on NQ-OPEN questions containing diverse types of ambiguity.
+We provide two distributions of our new dataset AmbigNQ: a full version with all annotation metadata and a light version with only inputs and outputs.
+"""
+_HOMEPAGE = "https://nlp.cs.washington.edu/ambigqa/"
+_LICENSE = "CC BY-SA 3.0"
+
+_URL = "https://nlp.cs.washington.edu/ambigqa/data/"
+_URLS = {
+    "light": _URL + "ambignq_light.zip",
+    "full": _URL + "ambignq.zip",
+}
+
+
+class AmbigQa(datasets.GeneratorBasedBuilder):
+    """AmbigQA dataset"""
+
+    VERSION = datasets.Version("1.0.0")
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="light",
+            version=VERSION,
+            description="AmbigNQ light version with only inputs and outputs",
+        ),
+        datasets.BuilderConfig(
+            name="full",
+            version=VERSION,
+            description="AmbigNQ full version with all annotation metadata",
+        ),
+    ]
+    DEFAULT_CONFIG_NAME = "full"
+
+    def _info(self):
+        features_dict = {
+            "id": datasets.Value("string"),
+            "question": datasets.Value("string"),
+            "annotations": datasets.features.Sequence(
+                {
+                    "type": datasets.Value("string"),  # datasets.ClassLabel(names = ["singleAnswer","multipleQAs"])
```

Comment:
It's fine this way since we use class label for classification labels in general :)